cherry_picker/cherry_picker/test.py

```diff
@@ -39,216 +52,241 @@ def changedir(d):
-                                                'prefix-22a594a-2.7',  # Not the prefix we were expecting
-                                                'backport-22a594a-base',  # No version info in the base branch
-                                                ]
-                         )
-@mock.patch('subprocess.check_output')
+@pytest.mark.parametrize(
+    "cherry_pick_branch",
+    [
+        "backport-22a594a",  # Not enough fields
+        "prefix-22a594a-2.7",  # Not the prefix we were expecting
+        "backport-22a594a-base",  # No version info in the base branch
+    ],
+)
+@mock.patch("subprocess.check_output")
 def test_get_base_branch_invalid(subprocess_check_output, cherry_pick_branch):
-    subprocess_check_output.return_value = b'22a594a0047d7706537ff2ac676cdc0f1dcb329c'
+    subprocess_check_output.return_value = b"22a594a0047d7706537ff2ac676cdc0f1dcb329c"
     with pytest.raises(ValueError):
         get_base_branch(cherry_pick_branch)
 
 
-@mock.patch('subprocess.check_output')
+@mock.patch("subprocess.check_output")
 def test_get_current_branch(subprocess_check_output):
-    subprocess_check_output.return_value = b'master'
-    assert get_current_branch() == 'master'
+    subprocess_check_output.return_value = b"master"
+    assert get_current_branch() == "master"
 
 
-@mock.patch('subprocess.check_output')
+@mock.patch("subprocess.check_output")
 def test_get_full_sha_from_short(subprocess_check_output):
     mock_output = b"""22a594a0047d7706537ff2ac676cdc0f1dcb329c"""
     subprocess_check_output.return_value = mock_output
-    assert get_full_sha_from_short('22a594a') == '22a594a0047d7706537ff2ac676cdc0f1dcb329c'
+    assert (
+        get_full_sha_from_short("22a594a") == "22a594a0047d7706537ff2ac676cdc0f1dcb329c"
+    )
 
 
-@mock.patch('subprocess.check_output')
+@mock.patch("subprocess.check_output")
 def test_get_author_info_from_short_sha(subprocess_check_output):
     mock_output = b"Armin Rigo <armin.rigo@gmail.com>"
     subprocess_check_output.return_value = mock_output
-    assert get_author_info_from_short_sha('22a594a') == 'Armin Rigo <armin.rigo@gmail.com>'
+    assert (
+        get_author_info_from_short_sha("22a594a") == "Armin Rigo <armin.rigo@gmail.com>"
+    )
 
 
-@pytest.mark.parametrize('input_branches,sorted_branches', [
-    (['3.1', '2.7', '3.10', '3.6'], ['3.10', '3.6', '3.1', '2.7']),
-    (['stable-3.1', 'lts-2.7', '3.10-other', 'smth3.6else'], ['3.10-other', 'smth3.6else', 'stable-3.1', 'lts-2.7']),
-])
-@mock.patch('os.path.exists')
+@pytest.mark.parametrize(
+    "input_branches,sorted_branches",
+    [
+        (["3.1", "2.7", "3.10", "3.6"], ["3.10", "3.6", "3.1", "2.7"]),
+        (
+            ["stable-3.1", "lts-2.7", "3.10-other", "smth3.6else"],
+            ["3.10-other", "smth3.6else", "stable-3.1", "lts-2.7"],
+        ),
+    ],
+)
+@mock.patch("os.path.exists")
+@mock.patch("cherry_picker.cherry_picker.validate_sha")
```

Comment:
Why?

Reply:
These tests aren't testing validate_sha itself, but the other functionalities. I think it is ok to be mocked.

Reply:
I disagree: more integrated tests is better. Testing a combination of things makes it all more robust. Especially when it costs nothing.
As in
![](https://cdn-images-1.medium.com/max/800/1*UtZzMT32fRMnSN-HmgiSVQ.gif)